src/deque.jl

```diff
@@ -196,3 +196,4 @@ end
 
 # TODO proper multidimension boundscheck
 @propagate_inbounds setindex(a::StaticArray, x, inds::Int...) = setindex(a, x, LinearIndices(a)[inds...])
+@propagate_inbounds setindex(a::StaticArray, x, I::CartesianIndex) = setindex(a, x, I.I...)
```

Comment:
I just looked at  `?CartesianIndex` and it says:
> One can freely mix integer and CartesianIndex indices;
  for example, A[Ipre, i, Ipost] (where Ipre and Ipost are CartesianIndex indices and i is an Int) can be a useful expression when writing algorithms that work along a single dimension of an array of
  arbitrary dimensionality.


```suggestion
@propagate_inbounds setindex(a::StaticArray, x, inds::Union{Integer, CartesianIndex}...) = setindex(a, x, LinearIndices(a)[inds...])
```